src/modules/cpu/kernel/phase.hpp

```diff
@@ -0,0 +1,965 @@
+                srcPtr1Row = srcPtr1Channel;
+                srcPtr2Row = srcPtr2Channel;
+                dstPtrRow = dstPtrChannel;
+
+                for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+                {
+                    Rpp32f *srcPtr1Temp, *srcPtr2Temp, *dstPtrTemp;
+                    srcPtr1Temp = srcPtr1Row;
+                    srcPtr2Temp = srcPtr2Row;
+                    dstPtrTemp = dstPtrRow;
+
+                    int vectorLoopCount = 0;
+                    for (; vectorLoopCount < alignedLength; vectorLoopCount += 8)
+                    {
+                        __m256 p1[1], p2[1];
+
+                        rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtr1Temp, p1);    // simd loads
+                        rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtr2Temp, p2);    // simd loads
+                        p1[0] = _mm256_mul_ps(atan2_ps(p1[0], p2[0]), pMul);    // phase computation
+                        rpp_simd_store(rpp_store8_f32_to_f32_avx, dstPtrTemp, p1);    // simd stores
+
+                        srcPtr1Temp += 8;
+                        srcPtr2Temp += 8;
+                        dstPtrTemp += 8;
+                    }
+                    for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                    {
+                        *dstPtrTemp++ = RPPPIXELCHECKF32(atan(*srcPtr1Temp / *srcPtr2Temp) * multiplier);
+
+                        srcPtr1Temp++;
+                        srcPtr2Temp++;
+                    }
+
+                    srcPtr1Row += srcDescPtr->strides.hStride;
+                    srcPtr2Row += srcDescPtr->strides.hStride;
+                    dstPtrRow += dstDescPtr->strides.hStride;
+                }
+
+                srcPtr1Channel += srcDescPtr->strides.cStride;
+                srcPtr2Channel += srcDescPtr->strides.cStride;
+                dstPtrChannel += dstDescPtr->strides.cStride;
+            }
+        }
+    }
+
+    return RPP_SUCCESS;
+}
+
+RppStatus phase_f16_f16_host_tensor(Rpp16f *srcPtr1,
+                                    Rpp16f *srcPtr2,
+                                    RpptDescPtr srcDescPtr,
+                                    Rpp16f *dstPtr,
+                                    RpptDescPtr dstDescPtr,
+                                    RpptROIPtr roiTensorPtrSrc,
+                                    RpptRoiType roiType,
+                                    RppLayoutParams layoutParams,
+                                    rpp::Handle& handle)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f multiplier = 1 / 1.570796;
```

Comment:
May be explain why this is the multipler? try and see if you can explain why this is the case

Reply:
Added comment in common.h file